### PR TITLE
Da.resize

### PR DESCRIPTION
**Pull request by aneksteind**

### CODE REVIEW

File: src/lcm_republisher.cpp [hunk 1]
```diff
@@ -165,6 +169,16 @@ class LCMRepublisher
       rgb_height = rgb_msg->height;
       depth_width = depth_msg->width;
       depth_height = depth_msg->height;
+
+      if (rgb_width != depth_width || rgb_height != depth_height) {
+        ROS_WARN("Expected rgb dimensions (%d,%d) to be equal to depth dimensions (%d,%d)",
+        rgb_width, rgb_height, depth_width, depth_height);
+      }
+
+      //$ user-specified outgoing image dimensions for both depth and rgb
```
Comment (5yler) [inline src/lcm_republisher.cpp:178]: Please don't add dollar signs at the start of comment, that's meant to indicate I wrote it haha

File: src/lcm_republisher.cpp [hunk 2]
```diff
@@ -394,7 +394,17 @@ class LCMRepublisher
...
+    if (enforce_resize_) {
+      cv::Mat resized_rgb;
+      cv::Mat resized_depth;
+
+      cv::resize(rgb, resized_rgb, cv::Size(resize_width_, resize_height_));
+      cv::resize(depth, resized_depth, cv::Size(resize_width_, resize_height_));
+
+      publishLCM((long unsigned int) cloud->header.stamp, resized_rgb, resized_depth);
+    } else {
+      publishLCM((long unsigned int) cloud->header.stamp, rgb, depth);
```
Comment (5yler) [inline src/lcm_republisher.cpp:416]: Could you just do 
``` 
cv::resize(rgb, rgb, cv::Size(resize_width_, resize_height_));
```
or does that not work?

File: launch/lcm_republisher.launch [hunk 3]
```diff
@@ -9,6 +9,9 @@
       rgb_topic: /camera/rgb/image_rect_color
       depth_topic: /camera/depth_registered/sw_registered/image_rect_raw
       cloud_topic: /camera/depth_registered/points
+      enforce_resize: true
+      resize_width: 1280
+      resize_height: 720
```
Comment (5yler) [inline launch/lcm_republisher.launch:14]: 640 x 480

File: launch/lcm_republisher.launch [hunk 4]
```diff
@@ -9,6 +9,9 @@
       rgb_topic: /camera/rgb/image_rect_color
       depth_topic: /camera/depth_registered/sw_registered/image_rect_raw
       cloud_topic: /camera/depth_registered/points
+      enforce_resize: true
```
Comment (5yler) [inline launch/lcm_republisher.launch:12]: Prob. default to false here

File: launch/lcm_republisher.launch (hunk 3 above)
Comment (5yler) [reply launch/lcm_republisher.launch:14]: Wait that math does not work

File: src/lcm_republisher.cpp (hunk 2 above)
Comment (aneksteind) [reply src/lcm_republisher.cpp:416]: I'm not aware whether or not opencv can do a resize in place, I need to do some more research on it

File: src/lcm_republisher.cpp (hunk 1 above)
Comment (aneksteind) [reply src/lcm_republisher.cpp:178]: Thought it was some special auto doc spec I had to adhere to :)

**Review by 5yler — APPROVED** [review]
LGTM platform & style

**Pull request merged**